inspect.go

```diff
@@ -0,0 +1,171 @@
+package enmime
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"mime"
+	"net/textproto"
+	"strings"
+
+	"github.com/jhillyerd/enmime/internal/coding"
+	"github.com/pkg/errors"
+)
+
+var defaultHeadersList = []string{
+	"From",
+	"To",
+	"Sender",
+	"CC",
+	"BCC",
+	"Subject",
+	"Date",
+}
+
+// DecodeHeaders returns a limited selection of mime headers for use by user agents
+// Default header list:
+//   "Date", "Subject", "Sender", "From", "To", "CC" and "BCC"
+//
+// Additional headers provided will be formatted canonically:
+//   h, err := enmime.DecodeHeaders(b, "content-type", "user-agent")
+func DecodeHeaders(b []byte, addtlHeaders ...string) (textproto.MIMEHeader, error) {
+	b = ensureHeaderBoundary(b)
+	tr := textproto.NewReader(bufio.NewReader(bytes.NewReader(b)))
+	headers, err := tr.ReadMIMEHeader()
+	switch errors.Cause(err) {
+	case nil, io.EOF:
+	// carry on, io.EOF is expected
+	default:
+		return nil, err
+	}
+	headerList := append(defaultHeadersList, addtlHeaders...)
```

Comment:
This append risks modifying `defaultHeadersList`, should copy it first.  Probably not thread-safe either.

Reply:
done